database-plugins/src/main/java/co/cask/hydrator/plugin/db/batch/sink/DBSink.java

```diff
@@ -149,7 +146,14 @@ void setColumns(List<String> columns) {
 
   private void setResultSetMetadata() throws Exception {
     dbManager.ensureJDBCDriverIsAvailable(driverClass);
-    Map<String, Integer> columnToType = new HashMap<>();
+    Map<String, Integer> columnToType = new TreeMap<> (
+            new Comparator<String>() {
```

Comment:
The `String` class has a case insensitive comparator `String.CASE_INSENSITIVE_ORDER`. Please use that instead.